code/game/objects/items/devices/flashlight.dm

```diff
@@ -204,7 +214,7 @@
 /obj/item/device/flashlight/flare/proc/Light(var/mob/user as mob)
 	if(user)
 		if(!isturf(user.loc))
-			to_chat(user, "You cannot turn the light on while in this [user.loc].")//To prevent some lighting anomalities.
+			to_chat(user, "You cannot light the flare while in this [user.loc].")//To prevent some lighting anomalities.
```

Comment:
You can completely remove this location check actually. It's an artifact from the old lighting system.